metadata-db-app/int_test/cmr/metadata_db/int_test/utility.clj

```diff
@@ -644,7 +644,9 @@
   [concept]
   (let [{:keys [concept-id revision-id]} concept
         stored-concept (:concept (get-concept-by-id-and-revision concept-id revision-id))]
-    (is (= (expected-concept concept) (dissoc stored-concept :revision-date :transaction-id :created-at)))))
+    (is (= (util/remove-nil-keys (expected-concept concept))
```

Comment:
Why did we need to make this change? What has nil values being returned?

Reply:
This change was from the previous direction we had for the ticket (which has since changed). In that (abandoned) implementation, a test was failing due to nil values being present when comparing expected services with returned services. This changed fixed those comparisons.

Since this would likely bite others in the future, I thought it good to keep it ... not so?

Reply:
This reminds me of a case I had before. I remember Leo didn't want me to remove the nil keys but to fix the cause of it.
That way, it will expose the problem when it occurs.  I have a mix feeling about this.

Reply:
Okay, I can remove this.

Reply:
Done.